h/static/scripts/controllers/search-bar-controller.js

```diff
@@ -0,0 +1,143 @@
+'use strict';
+
+var Controller = require('../base/controller');
+var setElementState = require('../util/dom').setElementState;
+
+/**
+ * Controller for the search bar.
+ */
+class SearchBarController extends Controller {
+  constructor(element) {
+    super(element);
+
+    this._input = this.refs.searchBarInput;
+    this._dropdown = this.refs.searchBarDropdown;
+    this._dropdownItems = element.querySelectorAll('[data-ref="searchBarDropdownItem"]');
+
+    var getActiveElementIndex = () => {
+      return Array.prototype.slice.call(this._dropdownItems).
+        indexOf(element.querySelector('.js-search-bar-dropdown-menu-item--active'));
```

Comment:
This confused me for a second (why not just do `this._dropdownItems.indexOf(...)`?) but I realised that it's because `this._dropdownItems` is a DOM `NodeList` not a JavaScript array, and `NodeList`s, don't have `.indexOf()`, and `Array.prototype.slice.call(this._dropdownItems)` converts the `NodeList` into an array.

With ES6 I think you can convert the `NodeList` into an array like this instead: `return [...this._dropdownItems].indexOf(...);`.

Also, I wonder if this should just be done at the top of the file when `this._dropdownItems` is first defined? Then it would always be an array and we wouldn't have to convert it in multiple places:

```javascript
this._dropdownItems = element.querySelectorAll('[data-ref="searchBarDropdownItem"]');
// Turn this._dropdownItems from a NodeList into an array:
this._dropdownItems = [...this._dropdownItems];
```